src/react_native/cameraroll.cljs

```diff
@@ -5,15 +5,29 @@
     [taoensso.timbre :as log]
     [utils.transforms :as transforms]))
 
+(defn promise-get-photos
+  [opts]
+  (-> (.getPhotos CameraRoll (clj->js opts))
+      (.then transforms/js->clj)))
+
+(defn promise-get-albums
+  [opts]
+  (-> (.getAlbums CameraRoll (clj->js opts))
+      (.then transforms/js->clj)))
+
+
+
 (defn get-photos
   [opts callback]
   (-> (.getPhotos CameraRoll (clj->js opts))
+      ;; With callback approach, the value isn't accessible from the returned promise.
       (.then #(callback (transforms/js->clj %)))
       (.catch #(log/warn "could not get camera roll photos" %))))
 
 (defn get-albums
   [opts callback]
   (-> (.getAlbums CameraRoll (clj->js opts))
+      ;; With callback approach, the value isn't accessible from the returned promise.
       (.then #(callback (transforms/js->clj %)))
       (.catch #(log/warn "could not get camera roll albums" %))))
```

Comment:
@ulisesmac here's the "callback approach" I mentioned, where the resolved value is passed to the callback, but not passed down the promise chain.